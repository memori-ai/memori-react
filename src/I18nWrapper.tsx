import React, { useEffect } from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from './i18n';
import { mathJaxConfig } from './helpers/utils';

export interface Props {
  children: React.ReactNode;
}

const I18nWrapper = ({ children }: Props) => {
  useEffect(() => {
    // @ts-ignore
    if (!window.MathJax) window.MathJax = mathJaxConfig;
  }, []);

  return (
    <I18nextProvider i18n={i18n}>
      {children}

      <script
        id="MathJax-script"
        async
        src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"
      ></script>
    </I18nextProvider>
  );
};

export default I18nWrapper;
