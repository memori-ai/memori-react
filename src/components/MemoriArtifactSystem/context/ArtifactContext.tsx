import React, { createContext, useContext, useState, ReactNode, useCallback, useMemo } from 'react';
import { ArtifactData, ArtifactSystemState } from '../types/artifact.types';

interface ArtifactContextType {
  state: ArtifactSystemState;
  openArtifact: (artifact: ArtifactData) => void;
  closeArtifact: () => void;
  toggleFullscreen: () => void;
}

const ArtifactContext = createContext<ArtifactContextType | null>(null);

export const ArtifactProvider = ({ children }: { children: ReactNode }) => {
  const [state, setState] = useState<ArtifactSystemState>({
    currentArtifact: null,
    isDrawerOpen: false,
    isFullscreen: false,
  });

  const openArtifact = useCallback((artifact: ArtifactData) => {
    setState(prev => {
      // Only update if the artifact is different
      if (prev.currentArtifact?.id === artifact.id && prev.isDrawerOpen) {
        return prev;
      }
      return {
        currentArtifact: artifact,
        isDrawerOpen: true,
        isFullscreen: false,
      };
    });
  }, []);

  const closeArtifact = useCallback(() => {
    setState(prev => {
      // if (!prev.isDrawerOpen) return prev;
      return {
        ...prev,
        currentArtifact: null,
        isDrawerOpen: false,
        isFullscreen: false,
      };
    });
  }, []);

  const toggleFullscreen = useCallback(() => {
    setState(prev => ({
      ...prev,
      isFullscreen: !prev.isFullscreen,
    }));
  }, []);

  const contextValue = useMemo(() => ({
    state,
    openArtifact,
    closeArtifact,
    toggleFullscreen,
  }), [state, openArtifact, closeArtifact, toggleFullscreen]);

  return (
    <ArtifactContext.Provider value={contextValue}>
      {children}
    </ArtifactContext.Provider>
  );
};

export const useArtifact = () => {
  const context = useContext(ArtifactContext);
  if (!context) {
    throw new Error('useArtifact must be used within ArtifactProvider');
  }
  return context;
};
