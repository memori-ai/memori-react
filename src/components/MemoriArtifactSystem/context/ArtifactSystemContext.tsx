/**
 * ArtifactSystemContext
 * React Context for sharing artifact system state and actions across the application
 */

import React, { createContext, useContext, ReactNode } from 'react';
import { 
  ArtifactSystemState, 
  ArtifactSystemActions, 
  ArtifactSystemConfig,
  ArtifactSystemHook 
} from '../types/artifact.types';
import { useArtifactSystem } from '../hooks/useArtifactSystem';

// Context type definition
interface ArtifactSystemContextType extends ArtifactSystemHook {
  // Additional context-specific properties can be added here if needed
}

// Create the context
const ArtifactSystemContext = createContext<ArtifactSystemContextType | null>(null);

// Provider props interface
interface ArtifactSystemProviderProps {
  children: ReactNode;
  config?: Partial<ArtifactSystemConfig>;
}

// Provider component
export const ArtifactSystemProvider: React.FC<ArtifactSystemProviderProps> = ({ 
  children, 
  config = {} 
}) => {
  const artifactSystem = useArtifactSystem(config);

  return (
    <ArtifactSystemContext.Provider value={artifactSystem}>
      {children}
    </ArtifactSystemContext.Provider>
  );
};

// Custom hook to use the artifact system context
export const useArtifactSystemContext = (): ArtifactSystemContextType => {
  const context = useContext(ArtifactSystemContext);
  
  if (!context) {
    throw new Error(
      'useArtifactSystemContext must be used within an ArtifactSystemProvider'
    );
  }
  
  return context;
};

// Export the context for advanced usage
export { ArtifactSystemContext };
