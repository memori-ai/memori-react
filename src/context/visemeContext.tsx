import React, {
  createContext,
  useContext,
  useRef,
  useCallback,
  useState,
  useMemo,
} from 'react';
import { IAudioContext } from 'standardized-audio-context';

type Viseme = {
  name: string;
  weight: number;
  startTime: number;
  endTime: number;
};

interface VisemeContextType {
  addViseme: (visemeId: number, audioOffset: number) => void;
  updateCurrentViseme: (currentTime: number) => Viseme | null;
  startProcessing: (audioCtx: IAudioContext) => void;
  stopProcessing: () => void;
  resetAndStartProcessing: (audioCtx: IAudioContext) => void;
  resetVisemeQueue: () => void;
  isProcessing: boolean;
  setAudioContext: (ctx: IAudioContext) => void;
}

const VisemeContext = createContext<VisemeContextType | undefined>(undefined);

const VISEME_MAP: Readonly<{ [key: number]: string }> = {
  0: 'viseme_sil', // silence
  1: 'viseme_PP', // p, b, m
  2: 'viseme_FF', // f, v
  3: 'viseme_TH', // th, dh
  4: 'viseme_DD', // t, d, n, l
  5: 'viseme_kk', // k, g, ng
  6: 'viseme_CH', // tS, dZ, S, Z
  7: 'viseme_SS', // s, z
  8: 'viseme_nn', // Not explicitly defined in Azure mapping, keeping for compatibility
  9: 'viseme_RR', // r
  10: 'viseme_aa', // A:
  11: 'viseme_E', // e
  12: 'viseme_I', // I
  13: 'viseme_O', // O
  14: 'viseme_U', // u
  15: 'viseme_kk', // g, k (same as 5)
  16: 'viseme_CH', // ch, j, sh, zh (same as 6)
  17: 'viseme_SS', // s, z (same as 7)
  18: 'viseme_TH', // th, dh (same as 3)
  19: 'viseme_RR', // r (same as 9)
  20: 'viseme_kk', // w (closest match, could be debated)
  21: 'viseme_PP', // y (closest match, could be debated)
};

// Optimized constants with explanations
const CONSTANTS = {
  // Duration of each viseme - optimized for natural speech
  // Slightly longer to ensure smoother transitions
  // Average syllable duration in natural speech is ~0.2s
  // We want shorter duration for more precise lip movements
  DEFAULT_VISEME_DURATION: 0.1, // 100ms per viseme

  // Overlap between consecutive visemes
  // Increased for smoother blending between mouth shapes
  // Should be about 1/3 of the viseme duration for natural transition
  VISEME_OVERLAP: 0.02, // 30ms overlap

  // Smoothing factor for weight transitions
  // Lower value = smoother but slower transitions
  // Higher value = faster but potentially jerky transitions
  // 0.4 provides good balance between responsiveness and smoothness
  SMOOTHING_FACTOR: 0.4,

  // How often to log debug information (in frames)
  // Adjusted to be less frequent to reduce console spam
  // 60 frames = roughly 1 second at 60fps
  LOG_INTERVAL: 60,

  // Time to preload/buffer visemes (in seconds)
  // Reduced from 1s to 0.5s as we don't need such a large buffer
  // Half a second is enough to prepare upcoming visemes
  PRELOAD: 0.5,

  // Weight multiplication factor for emphasis
  // Can be adjusted to make lip movements more or less pronounced
  WEIGHT_MULTIPLIER: 0.8,
} as const;

// Export individual constants for backwards compatibility
export const {
  DEFAULT_VISEME_DURATION,
  VISEME_OVERLAP,
  SMOOTHING_FACTOR,
  LOG_INTERVAL,
  PRELOAD,
  WEIGHT_MULTIPLIER,
} = CONSTANTS;

// Helper functions for timing calculations
const timing = {
  // Calculate viseme end time based on start time
  calculateEndTime: (startTime: number) => startTime + DEFAULT_VISEME_DURATION,

  // Check if viseme is expired
  isExpired: (viseme: Viseme, currentTime: number) =>
    viseme.endTime < currentTime - VISEME_OVERLAP,

  // Calculate viseme weight based on progress
  calculateWeight: (progress: number) =>
    Math.sin(Math.PI * Math.min(progress, 1)) * WEIGHT_MULTIPLIER,

  // Calculate smooth transition between weights
  smoothWeight: (currentWeight: number, targetWeight: number) =>
    currentWeight + (targetWeight - currentWeight) * SMOOTHING_FACTOR,
};

const createLogger =
  (type: 'event' | 'error' | 'debug') => (event: string, data: any) => {
    const styles = {
      event: 'color: #4CAF50; font-weight: bold;',
      error: 'color: #f44336; font-weight: bold;',
      debug: 'color: #2196F3; font-weight: bold;',
    };
    console.log(`%c[VisemeContext] ${event}`, styles[type], data);
  };

const logVisemeEvent = createLogger('event');
const logVisemeError = createLogger('error');
const logVisemeDebug = createLogger('debug');

type VisemeState = 'idle' | 'preparing' | 'active' | 'paused' | 'finished';

export const VisemeProvider: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const visemeQueueRef = useRef<Viseme[]>([]);
  const audioContextRef = useRef<IAudioContext | null>(null);
  const audioStartTimeRef = useRef<number | null>(null);
  const [isProcessing, setIsProcessing] = useState(false);
  const [visemeState, setVisemeState] = useState<VisemeState>('idle');
  const lastVisemeRef = useRef<Viseme | null>(null);
  const frameCountRef = useRef(0);
  const firstVisemeTimeRef = useRef<number | null>(null);

  const setAudioContext = useCallback((ctx: IAudioContext) => {
    audioContextRef.current = ctx;

    // Listen to audio context state changes
    ctx.onstatechange = () => {
      logVisemeEvent('Audio Context State Change', {
        state: ctx.state,
        currentTime: ctx.currentTime,
      });

      switch (ctx.state) {
        case 'running':
          setVisemeState('active');
          break;
        case 'suspended':
          setVisemeState('paused');
          break;
        case 'closed':
          setVisemeState('finished');
          stopProcessing();
          break;
      }
    };
  }, []);

  const addViseme = useCallback(
    (visemeId: number, audioOffset: number) => {
      if (visemeState === 'finished') return;

      const visemeName = VISEME_MAP[visemeId] || 'viseme_sil';
      const startTime = audioOffset / 10000000;

      // Store first viseme time as reference
      if (firstVisemeTimeRef.current === null) {
        firstVisemeTimeRef.current = startTime;
      }

      // Calculate time relative to first viseme
      const relativeStartTime = startTime - (firstVisemeTimeRef.current || 0);
      const endTime = relativeStartTime + DEFAULT_VISEME_DURATION;

      const newViseme: Viseme = {
        name: visemeName,
        weight: 0,
        startTime: relativeStartTime,
        endTime,
      };

      visemeQueueRef.current.push(newViseme);

      if (visemeState === 'idle') {
        setVisemeState('preparing');
      }

      // logVisemeEvent('Added Viseme', {
      //   visemeId,
      //   name: visemeName,
      //   absoluteStartTime: startTime,
      //   relativeStartTime,
      //   endTime,
      //   queueLength: visemeQueueRef.current.length,
      //   state: visemeState,
      // });
    },
    [visemeState]
  );

  const startProcessing = useCallback((audioCtx: IAudioContext) => {
    if (!audioCtx) {
      logVisemeError('No audio context provided', { state: visemeState });
      return;
    }

    audioContextRef.current = audioCtx;
    audioStartTimeRef.current = audioCtx.currentTime;
    frameCountRef.current = 0;
    setIsProcessing(true);
    setVisemeState('active');

    logVisemeEvent('Started Processing', {
      audioTime: audioCtx.currentTime,
      queueLength: visemeQueueRef.current.length,
      state: visemeState,
    });
  }, []);

  const stopProcessing = useCallback(() => {
    setIsProcessing(false);
    setVisemeState('finished');
    audioStartTimeRef.current = null;
    lastVisemeRef.current = null;
    frameCountRef.current = 0;
    audioContextRef.current = null;

    logVisemeEvent('Stopped Processing', {
      queueLength: visemeQueueRef.current.length,
      state: visemeState,
    });
  }, []);

  const updateCurrentViseme = useCallback(
    (_: number): Viseme | null => {
      if (!isProcessing || !audioContextRef.current) return null;

      const audioTime =
        audioContextRef.current.currentTime -
        (audioStartTimeRef.current || 0) +
        PRELOAD;

      // Remove expired visemes
      visemeQueueRef.current = visemeQueueRef.current.filter(
        v => !timing.isExpired(v, audioTime)
      );

      // Find current active viseme
      const currentViseme = visemeQueueRef.current.find(
        v => v.startTime <= audioTime && v.endTime > audioTime - VISEME_OVERLAP
      );

      //log it every LOG_INTERVAL frames
      if (frameCountRef.current % LOG_INTERVAL === 60) {
        logVisemeDebug('Current Viseme', {
          currentViseme,
          audioTime,
          visemeQueue: visemeQueueRef.current,
        });
      }

      if (currentViseme) {
        const progress =
          (audioTime - currentViseme.startTime) /
          (currentViseme.endTime - currentViseme.startTime);

        const targetWeight = timing.calculateWeight(progress);

        const smoothedWeight = lastVisemeRef.current
          ? timing.smoothWeight(lastVisemeRef.current.weight, targetWeight)
          : targetWeight;

        const updatedViseme = { ...currentViseme, weight: smoothedWeight };
        lastVisemeRef.current = updatedViseme;

        return updatedViseme;
      }

      // Handle fade out
      if (lastVisemeRef.current) {
        const reducedWeight =
          lastVisemeRef.current.weight * (1 - SMOOTHING_FACTOR);
        lastVisemeRef.current = {
          ...lastVisemeRef.current,
          weight: reducedWeight,
        };
      }

      lastVisemeRef.current = null;
      return null;
    },
    [isProcessing]
  );

  const resetVisemeQueue = useCallback(() => {
    visemeQueueRef.current = [];
    lastVisemeRef.current = null;
    audioStartTimeRef.current = null;
    firstVisemeTimeRef.current = null;
    frameCountRef.current = 0;
    setVisemeState('idle');

    logVisemeEvent('Reset Viseme Queue', {
      previousState: visemeState,
    });
  }, [visemeState]);

  const resetAndStartProcessing = useCallback(
    (audioCtx: IAudioContext) => {
      logVisemeEvent('Reset And Start Processing', {
        previousState: visemeState,
        queueLength: visemeQueueRef.current.length,
      });

      stopProcessing();
      resetVisemeQueue();
      startProcessing(audioCtx);
    },
    [stopProcessing, resetVisemeQueue, startProcessing, visemeState]
  );

  const contextValue = useMemo(
    () => ({
      addViseme,
      updateCurrentViseme,
      startProcessing,
      stopProcessing,
      resetAndStartProcessing,
      resetVisemeQueue,
      isProcessing,
      setAudioContext,
    }),
    [
      addViseme,
      updateCurrentViseme,
      startProcessing,
      stopProcessing,
      resetAndStartProcessing,
      resetVisemeQueue,
      isProcessing,
      setAudioContext,
    ]
  );

  return (
    <VisemeContext.Provider value={contextValue}>
      {children}
    </VisemeContext.Provider>
  );
};

export const useViseme = () => {
  const context = useContext(VisemeContext);
  if (!context) {
    throw new Error('useVisemeContext must be used within a VisemeProvider');
  }
  return context;
};
