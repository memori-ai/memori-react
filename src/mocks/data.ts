import {
  Medium,
  Memori,
  Message,
  Memory,
  Integration,
  Tenant,
  DialogState,
  KnownFact,
  ExpertReference,
  User,
  Venue,
} from '@memori.ai/memori-api-client/dist/types';

export const sessionID = '131165be-9d1a-42fb-a3ce-e8f86d40c88f';

export const tenant: Tenant = {
  adminCount: 3,
  config: {
    name: 'AIsuru',
    requirePosition: false,
    showNewUser: true,
  },
  creationTimestamp: '2023-05-31T14:32:48.885287Z',
  description: 'AIsuru',
  disableRegistration: false,
  id: 'www.aisuru.com',
  lastChangeTimestamp: '2023-05-31T14:32:48.885287Z',
  logoURL: 'https://aisuru.com/images/aisuru/logo.png',
  maxAdmins: 0,
  maxCompletions: 0,
  maxCompletionsPerUser: 0,
  maxFreeSessions: 400,
  maxFreeSessionsPerUser: 100,
  maxMemoriPerAdmin: 0,
  maxMemoriPerUser: 3,
  maxTotalMemori: 0,
  maxUsers: 0,
  memoriCount: 0,
  name: 'www.aisuru.com',
  nonFreeSessionCost: 0.02,
  paying: true,
  tenantID: '96caa4b4-31a4-48e5-8163-dec61869a2a7',
  theme: 'aisuru',
  userCount: 0,
};

export const user: User = {
  tenant: 'localhost:3000',
  userID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  userName: 'nzambello',
  eMail: 'nicola@nzambello.dev',
  admin: false,
  maxMemori: 0,
  enableMemoriCreation: true,
  enableBoardOfExperts: true,
  maxFreeSessions: 0,
  tnCAndPPAccepted: true,
  tnCAndPPAcceptanceDate: '2021-03-01T00:00:00.000Z',
  pAndCUAccepted: true,
  pAndCUAcceptanceDate: '2021-03-01T00:00:00.000Z',
  birthDate: '1900-03-01T00:00:00.000Z',
  age: 28,
  avatarURL: 'https://avatars.githubusercontent.com/u/21101435?v=4',
};

export const memori: Memori = {
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  engineMemoriID: '66b4e161-2431-4b21-9b70-d8c27de730ca',
  name: 'Memori',
  memoriConfigurationID: 'MemoriCloud-it_IT',
  description: 'Lorem ipsum.',
  voiceType: 'male',
  isGiver: true,
  isReceiver: false,
  privacyType: 'PUBLIC',
  needsPosition: false,
  culture: 'it-IT',
  categories: [],
  publishedInTheMetaverse: true,
  exposed: true,
  enableCompletions: true,
  nsfw: false,
  ageRestriction: 14,
  contentQualityIndex: 66.6,
  contentQualityIndexTimestamp: '2021-03-01T12:00:00.000Z',
  ownerUserName: 'username',
  ownerTenantName: 'aisuru.com',
  ownerUserID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  metaverseEnvironment: 'synthwave',
  secretToken: 'awanagana',
  giverPIN: '1234',
  giverTag: 'üßë‚Äçüíª',
  avatarURL: 'https://aisuru.com/images/aisuru/square_logo.png',
  coverURL: 'https://aisuru.com/images/aisuru/og-image.png',
};

export const history: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const historyWithMedia: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui delle cose per te.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai/en',
        mimeType: 'text/html',
        title: 'Link Memori Srl',
      },
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efce',
        url: 'https://rawmaterial.it/en',
        mimeType: 'text/html',
        title: 'Link RawMaterial',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
        mimeType: 'text/html',
        title: 'Introducing Plone Remix | Vimeo',
        url: 'https://vimeo.com/766468314',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5969',
        mimeType: 'text/html',
        title: 'A sustainable web: is it possible? - Nicola Zambello | YouTube',
        url: 'https://www.youtube.com/watch?v=feH26j3rBz8',
      },
      ...Array.from({ length: 3 }, (_, i) => ({
        mediumID: `95226d7e-7bae-465e-8b80-995587bb597${i}`,
        mimeType: 'image/png',
        title: `Image ${i}`,
        url: `https://picsum.photos/${i % 2 ? '200' : '300'}/${
          i % 3 ? '300' : '200'
        }?random=${i}`,
      })),
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
];

export const historyWithAIGeneratedMessages: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Sa. Sa. Prova',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    generatedByAI: true,
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const dialogState: DialogState = {
  state: 'R1',
  previousState: 'I0',
  stateName: 'WaitingForReceiverQuestion',
  confidence: 1,
  knownTags: {
    '‚ò†Ô∏è': 'test',
    'üòé': 'Ciccio',
  },
  emission: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
  hints: ['Va bene', 'No grazie'],
  media: [],
  acceptsTimeout: true,
  acceptsAbort: false,
  acceptsMedia: false,
  acceptsDate: false,
  acceptsPlace: false,
  acceptsTag: false,
  giverID: 'c832e2dc-403c-4baf-a3b7-2374e100dbcf',
  contextVars: {},
};

export const medium: Medium = {
  mediumID: 'a196b513-d745-4121-8913-8f457b999add',
  url: 'https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2',
  mimeType: 'image/jpeg',
  title: 'Game Cover',
  creationTimestamp: '2022-05-03T08:57:30.584439Z',
  creationName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
  lastChangeTimestamp: '2022-05-03T14:19:09.932865Z',
  lastChangeName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
};

export const integration: Integration = {
  integrationID: 'cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1',
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  type: 'LANDING_EXPERIENCE',
  state: 'NEW',
  deviceEmails: [],
  customData:
    '{"textColor":"#2a2a2a","buttonBgColor":"#823ce1","buttonTextColor":"#ffffff","name":"Web","globalBackground":"https://assets.memori.ai/api/v2/asset/364e498c-11da-42d5-9e32-19e5d137d4b8.jpeg","blurBackground":true,"innerBgColor":"light","innerBgAlpha":0.8,"multilanguage":true,"avatar":"readyplayerme","avatarURL":"https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb#1669663599444"}',
  resources: [],
  publish: true,
  creationTimestamp: '2022-06-11T14:13:45.685038Z',
  lastChangeTimestamp: '2022-06-11T14:13:45.685038Z',
};

export const knownFact: KnownFact = {
  knownFactID: 'b0b0b0b0-b0b0-b0b0-b0b0-b0b0b0b0b0b3',
  knownFactType: 'ShortTerm',
  text: 'I am a known fact. Quisque in ultrices lectus. Nulla at urna diam. Proin sodales lobortis libero eu facilisis.',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const expertReference: ExpertReference = {
  expertID: 'f016f204-c307-483b-8891-680a3c974c53',
  name: 'TEST MEMORI',
  description: 'lui √® competente',
  default: true,
  expertMemoriID: 'dfc44f1c-1ba7-4e1e-a234-28aa8b6b3d32',
  expertBaseURL: 'http://localhost:7778',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const venue: Venue = {
  latitude: 44.66579,
  longitude: 11.48823,
  placeName: 'Altedo, Bologna, Italy',
  uncertainty: 2,
};

export const memoryQuestion: Memory = {
  memoryID: '08c0697d-f7f3-4a47-9970-aff75f01fb6c',
  memoryType: 'Question',
  lastRead: undefined,
  readOccurrences: undefined,
  receiverID: undefined,
  receiverTag: undefined,
  receiverName: undefined,
  media: [
    {
      mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
      mimeType: 'text/html',
      url: 'https://rawmaterial.it/it',
      title: 'RawMaterial',
    },
  ],
  title: 'sei umano',
  titleVariants: ['sei reale'],
  answers: [
    {
      text: 'se ti dico si ci credi?',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:25.410377',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:25.410406',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
    {
      text: 'probabilmente lo sono pi√π di te',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:36.957656',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:36.957657',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
  ],
  date: undefined,
  dateUncertaintyDays: undefined,
  placeName: undefined,
  placeLatitude: undefined,
  placeLongitude: undefined,
  placeUncertaintyKm: undefined,
  preformatted: false,
  conclusive: false,
  notPickable: false,
  contextVarsToSet: {
    SUBJECT: 'TRAVEL',
  },
  contextVarsToMatch: {
    SUBJECT: 'TRANSPORTS',
  },
  creationTimestamp: '2022-03-23T09:37:16.568149',
  creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
  lastChangeTimestamp: '2022-03-23T09:37:36.957695',
  lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
};
