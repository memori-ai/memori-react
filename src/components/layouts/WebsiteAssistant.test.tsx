import { render } from '@testing-library/react';
import Memori from '../MemoriWidget/MemoriWidget';
import { integration, memori, tenant } from '../../mocks/data';
import I18nWrapper from '../../I18nWrapper';
import { VisemeProvider } from '../../context/visemeContext';
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

it('renders WEBSITE_ASSISTANT layout unchanged', () => {
  const { container } = render(
    <I18nWrapper>
      <VisemeProvider>
        <Memori
          showShare={true}
        showSettings={true}
        memori={memori}
        tenant={tenant}
        tenantID="aisuru.com"
        integration={integration}
          layout="WEBSITE_ASSISTANT"
        />
      </VisemeProvider>
    </I18nWrapper>
  );
  expect(container).toMatchSnapshot();
});
