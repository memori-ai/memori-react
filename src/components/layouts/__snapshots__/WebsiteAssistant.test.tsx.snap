// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WEBSITE_ASSISTANT layout unchanged 1`] = `
<div>
  <div
    class="memori memori-widget memori-layout-website_assistant memori-controls-bottom memori--avatar-readyplayerme memori--with-integration"
    data-memori-engine-state="{}"
    data-memori-id="66b4e161-2431-4b21-9b70-d8c27de730ca"
    data-memori-integration="cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1"
    data-memori-name="Memori"
    data-memori-secondary-id="25ced51c-3520-41af-8bbe-222d861b8e32"
    style="height: 100vh;"
  >
    <div
      class="memori-website_assistant--collapsed"
    >
      <div
        class="memori-website_assistant--trigger"
      >
        <button
          class="memori-button memori-button--ghost memori-button--circle memori-button--padded memori-website_assistant--trigger-button"
          title="Espandi"
        >
          <div
            class="memori-blob"
          >
            <figure>
              <img
                alt=""
                role="presentation"
                src="https://aisuru.com/images/aisuru/square_logo.png"
              />
            </figure>
            <div
              class="mainDiv"
            />
            <div
              class="mainDiv"
            />
            <div
              class="mainDiv"
            />
            <div
              class="mainDiv"
            />
            <div
              class="mainDiv"
            />
          </div>
        </button>
      </div>
    </div>
    <audio
      id="memori-audio"
      src="https://aisuru.com/intro.mp3"
      style="display: none;"
    />
  </div>
  <script
    async=""
    id="MathJax-script"
    src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"
  />
</div>
`;
